engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
Using the `getCanonicalHostName()` approach also has a couple of issues:

- it is also prone to misconfiguration since it uses the host DNS resolver configuration (/etc/resolve.conf ?) which may have wrong or incomplete configuration e.g. it should specify the correct local domain and/or correct nameservers to use

- it uses the reverse DNS lookup (IP -> host) which works only if those zones are configured in the DNS and returns correct info.

I thought it is rare and a bit odd for a host to determine its own FQDN using configured nameservers since they are mostly used for forward lookup (host -> IP) for remote hosts. Curious to know the configuration and setup used to test this fix.